### ISSUE TITLE
Support metric and loss loging on step

### ISSUE THREAD

**Issue by Hrovatin** [scverse/scvi-tools#3020]
In large datasets, the training may take only 1-2 epochs, meaning that the loss profile should be inspected on the step. Thus, it would be great to add an option to log losses and metrics on steps instead of epochs. I would suggest this for both training and validation losses, as in some pathological cases where there was sth wrong with the data setup, the training and validation losses actually did not match.

**Comment by canergen** [comment]
Can you create a PR for this (can be in just the regular Training code)? Do you have some benchmarking for training time to add validation loss at each step?

**Comment by Hrovatin** [comment]
I dont have benchmarks for the validation set - maybe getting just one batch of validation set would be good to speed it up as a proxy?

**Comment by ori-kron-wis** [comment]
Added
